centreon/www/front_src/src/Resources/Actions/Resource/Check/index.tsx

```diff
@@ -0,0 +1,193 @@
+import IconCheck from '@mui/icons-material/Sync';
+
+import { postData, useRequest, useSnackbar } from '@centreon/ui';
+
+import { Resource } from '../../../models';
+import {
+  labelCheck,
+  labelCheckCommandSent,
+  labelForcedCheck,
+  labelForcedCheckCommandSent
+} from '../../../translatedLabels';
+import { checkEndpoint } from '../../api/endpoint';
+import useAclQuery from '../aclQuery';
+import ResourceActionButton from '../ResourceActionButton';
+
+import { adjustedCheckedResources } from './helpers';
+import ListCheckOptions from './ListCheckOptions';
+
+interface DisplaySelectedOption {
+  checked: boolean;
+  forcedChecked: boolean;
+}
+
+interface Props {
+  selectedResources: Array<Resource>;
+  setSelectedResources: (update: SetStateAction<Array<Resource>>) => void;
+  testId: string;
+}
+
+const CheckActionButton = ({
+  selectedResources,
+  setSelectedResources,
+  testId
+}: Props): JSX.Element => {
+  const { t } = useTranslation();
+
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const [displaySelectedOption, setDisplaySelectedOption] =
+    useState<DisplaySelectedOption | null>(null);
+  const { sendRequest: checkResource } = useRequest({ request: postData });
+
+  const { canForcedCheck, canCheck } = useAclQuery();
+  const { showSuccessMessage } = useSnackbar();
+
+  const disableCheck = !canCheck(selectedResources);
+  const disableForcedCheck = !canForcedCheck(selectedResources);
+  const hasSelectedResources = selectedResources.length > 0;
+
+  const isCheckPermitted = canCheck(selectedResources) || !hasSelectedResources;
+  const isForcedCheckPermitted =
+    canForcedCheck(selectedResources) || !hasSelectedResources;
+
+  const displayPopover = (event: React.MouseEvent<HTMLElement>): void => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const closePopover = (): void => {
+    setAnchorEl(null);
+  };
+
+  const handleForcedCheck = (): void => {
+    setDisplaySelectedOption({ checked: false, forcedChecked: true });
+    setAnchorEl(null);
+    if (selectedResources.length <= 0) {
+      return;
+    }
+
+    checkResource({
+      data: {
+        check: { is_forced: true },
+        resources: adjustedCheckedResources({ resources: selectedResources })
+      },
+      endpoint: checkEndpoint
+    }).then(() => {
+      setSelectedResources([]);
+      showSuccessMessage(t(labelForcedCheckCommandSent));
+    });
+  };
+
+  const handleCheck = (): void => {
+    setDisplaySelectedOption({ checked: true, forcedChecked: false });
+    setAnchorEl(null);
+    if (selectedResources.length <= 0) {
+      return;
+    }
+
+    checkResource({
+      data: {
+        check: { is_forced: false },
+        resources: adjustedCheckedResources({ resources: selectedResources })
+      },
+      endpoint: checkEndpoint
+    }).then(() => {
+      setSelectedResources([]);
+      showSuccessMessage(t(labelCheckCommandSent));
+    });
+  };
```

Comment:
I do think you merge those functions and add parameters to handle check or force check